src/history/test/HistoryTests.cpp

```diff
@@ -585,88 +612,102 @@ TEST_CASE("History catchup", "[history][catchup][acceptance]")
     // megabytes of unnecessary log entries
     CatchupSimulation catchupSimulation{VirtualClock::REAL_TIME};
     auto checkpointLedger = catchupSimulation.getLastCheckpointLedger(3);
-    auto app = catchupSimulation.createCatchupApplication(
-        std::numeric_limits<uint32_t>::max(), Config::TESTDB_ON_DISK_SQLITE,
-        "app");
 
-    auto offlineNonCheckpointDestinationLedger =
-        checkpointLedger -
-        app->getHistoryManager().getCheckpointFrequency() / 2;
+    auto f = [&](bool useBucketListDB) {
```

Comment:
Similarly here, we don't need to test the new flag for each catchup trigger condition (this test mostly focuses on how catchup is triggered, so independent of the new flag). These tests already take really long time to run, so we only want to test the cases where the flag actually matters.

Reply:
This applies to other tests in this file, "History prefix catchup", "History catchup with different modes" etc. 

 I recommend writing just one new test, which trigger basic offline catchup with bucket application - this should be sufficient to test out the flow with BucketsDB. In this new test, just in case we should also have two versions of the app publishing original checkpoints, one with the flag and the other without. This way we'd also verify that publish flow is unaffected by the flag.

Reply:
actually, publish isn't touched by the bucketlist DB changes right? So maybe that change isn't necessary

Reply:
Correct, I'll just add the one test without publish queue.